src/lib/operators/sort.cpp

```diff
@@ -33,19 +20,77 @@ const std::string& Sort::name() const {
 std::shared_ptr<AbstractOperator> Sort::_on_deep_copy(
     const std::shared_ptr<AbstractOperator>& copied_input_left,
     const std::shared_ptr<AbstractOperator>& copied_input_right) const {
-  return std::make_shared<Sort>(copied_input_left, _column_id, _order_by_mode, _output_chunk_size);
+  return std::make_shared<Sort>(copied_input_left, _sort_definitions, _output_chunk_size);
 }
 
 void Sort::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
 std::shared_ptr<const Table> Sort::_on_execute() {
-  _impl = make_unique_by_data_type<AbstractReadOnlyOperatorImpl, SortImpl>(
-      input_table_left()->column_data_type(_column_id), input_table_left(), _column_id, _order_by_mode,
-      _output_chunk_size);
-  return _impl->_on_execute();
+  _sort_definition_data_types.resize(_sort_definitions.size());
+  std::transform(
+      _sort_definitions.begin(), _sort_definitions.end(), _sort_definition_data_types.begin(),
+      [this](const auto sort_definition) { return input_table_left()->column_data_type(sort_definition.column); });
+
```

Comment:
I'm not a big fan of transform. Simply iterating over the column ids and writing into the _sort_definitions vector would probably be more intuitive and might even be shorter.

Reply:
So basically like we had it before? https://github.com/hyrise/hyrise/pull/2005#discussion_r379437147

Reply:
Hmpf. Didn't see that. Another example of code reviews not being an exact science. :/

Personally, I don't see a big advantage on the side of `std::transform` (other than std::execution, which we won't use anytime soon). On the other hand, I find it more natural to read the `for` loop: "Iterate over all columns and write them to this vector". I'm done with reading the for loop by the time it takes me to recall the order of arguments for `std::transform` and I suspect that upcoming student generations, who are not familiar with the STL, will feel the same.